languageServer/src/features/StageCompletions.ts

```diff
@@ -0,0 +1,26 @@
+import {
+  CompletionItem,
+  CompletionItemKind,
+  TextDocumentPositionParams
+} from 'vscode-languageserver/node'
+import { BaseFeature } from './BaseFeature'
+
+export class StageCompletions extends BaseFeature {
+  protected setup(): void {
+    this.connection?.onCompletion(
+      (params: TextDocumentPositionParams): CompletionItem[] => {
+        const stageNames = this.query({
+          dvcYamls: '$.workspace[?(@.uri.match(/dvc.yaml$/))]',
+          stages: '$.dvcYamls[*].parsed.stages.*~'
```

Comment:
"Then for every dvcYaml it could find, get the parsed contents, look up the stages property and get their names (the ~ is to get the property names only)"